python/semantic_kernel/memory/storage/chroma_data_store.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from datetime import datetime
+from typing import TYPE_CHECKING, List, Optional
+
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.storage.data_entry import DataEntry
+from semantic_kernel.memory.storage.data_store_base import DataStoreBase
+
+if TYPE_CHECKING:
+    import chromadb
+    import chromadb.config
+    from chromadb.api.models.Collection import Collection
+    from chromadb.api.types import QueryResult
+
+
+def camel_to_snake(camel_str):
+    snake_str = camel_str[0].lower()
+    for char in camel_str[1:]:
+        if char.isupper():
+            snake_str += "_" + char.lower()
+        else:
+            snake_str += char
+    return snake_str
+
+
+class ChromaDataStore(DataStoreBase):
+    """
+    ChromaDataStore provides an interface to store and retrieve data using ChromaDB.
+    ChromaDataStore does not use ChromaDB's built-in embedding function,
+    as embeddings are directly provided from SemanticTextMemory.
+    Collection names with uppercase characters are not supported by ChromaDB, they will be automatically converted.
+
+    Args:
+        persist_directory (Optional[str], optional): Path to the directory where data will be persisted.
+            Defaults to None, which means the default settings for ChromaDB will be used.
+        client_settings (Optional["chromadb.config.Settings"], optional): A Settings instance to configure
+            the ChromaDB client. Defaults to None, which means the default settings for ChromaDB will be used.
+
+    Example:
+        # Create a ChromaDataStore with a local specified directory for data persistence
+        chroma_local_data_store = ChromaDataStore(persist_directory='/path/to/persist/directory')
+
+        # Create a ChromaDataStore with a custom Settings instance
+        chroma_remote_data_store = ChromaDataStore(
+            client_settings=Settings(
+                chroma_api_impl="rest",
+                chroma_server_host="xxx.xxx.xxx.xxx",
+                chroma_server_http_port="8000"
+            )
+        )
+    """
+
+    def __init__(
+        self,
+        persist_directory: Optional[str] = None,
+        client_settings: Optional["chromadb.config.Settings"] = None,
+    ) -> None:
+        try:
+            import chromadb
+            import chromadb.config
+
+        except ImportError:
+            raise ValueError(
+                "Could not import chromadb python package. "
+                "Please install it with `pip install chromadb`."
+            )
```

Comment:
@alexchaomander i think it is enough!

Reply:
I agree that an import error is adequate here.